diff --git a/AmbaConfig b/AmbaConfig
index e4d2e80..4895952 100644
--- a/AmbaConfig
+++ b/AmbaConfig
@@ -4,12 +4,29 @@
 ## History:
 ##    2013/06/05 - [Qian Shen]
 ##
-## Copyright (C) 2012-2016, Ambarella, Inc.
+## Copyright (c) 2016 Ambarella, Inc.
 ##
-## All rights reserved. No Part of this file may be reproduced, stored
-## in a retrieval system, or transmitted, in any form, or by any means,
-## electronic, mechanical, photocopying, recording, or otherwise,
-## without the prior consent of Ambarella, Inc.
+## This file and its contents ("Software") are protected by intellectual
+## property rights including, without limitation, U.S. and/or foreign
+## copyrights. This Software is also the confidential and proprietary
+## information of Ambarella, Inc. and its licensors. You may not use, reproduce,
+## disclose, distribute, modify, or otherwise prepare derivative works of this
+## Software or any portion thereof except pursuant to a signed license agreement
+## or nondisclosure agreement with Ambarella, Inc. or its authorized affiliates.
+## In the absence of such an agreement, you agree to promptly notify and return
+## this Software to Ambarella, Inc.
+##
+## THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+## INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF NON-INFRINGEMENT,
+## MERCHANTABILITY, AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+## IN NO EVENT SHALL AMBARELLA, INC. OR ITS AFFILIATES BE LIABLE FOR ANY DIRECT,
+## INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+## (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+## LOSS OF USE, DATA, OR PROFITS; COMPUTER FAILURE OR MALFUNCTION; OR BUSINESS
+## INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+## CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+## ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+## POSSIBILITY OF SUCH DAMAGE.
 ##
 
 config BUILD_AMBARELLA_UTILITIES_PACKAGE
diff --git a/log.c b/log.c
index 47f525c..cef1d85 100644
--- a/log.c
+++ b/log.c
@@ -4,14 +4,32 @@
  * History:
  *  Jun 5, 2013 - [qianshen] created file
  *
- * Copyright (C) 2012-2016, Ambarella ShangHai Co,Ltd
+ * Copyright (c) 2016 Ambarella, Inc.
  *
- * All rights reserved. No Part of this file may be reproduced, stored
- * in a retrieval system, or transmitted, in any form, or by any means,
- * electronic, mechanical, photocopying, recording, or otherwise,
- * without the prior consent of Ambarella
+ * This file and its contents ( "Software" ) are protected by intellectual
+ * property rights including, without limitation, U.S. and/or foreign
+ * copyrights. This Software is also the confidential and proprietary
+ * information of Ambarella, Inc. and its licensors. You may not use, reproduce,
+ * disclose, distribute, modify, or otherwise prepare derivative works of this
+ * Software or any portion thereof except pursuant to a signed license agreement
+ * or nondisclosure agreement with Ambarella, Inc. or its authorized affiliates.
+ * In the absence of such an agreement, you agree to promptly notify and return
+ * this Software to Ambarella, Inc.
+ *
+ * THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF NON-INFRINGEMENT,
+ * MERCHANTABILITY, AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL AMBARELLA, INC. OR ITS AFFILIATES BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; COMPUTER FAILURE OR MALFUNCTION; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
  *
  ******************************************************************************/
+
 #include <stdio.h>
 #include <sys/types.h>
 #include <fcntl.h>
@@ -29,17 +47,17 @@ typedef enum {
 	LOG_TARGET_FILE,
 } LOG_TARGET;
 
-static LOG_LEVEL log_level = LOG_INFO;
+static AMBA_LOG_LEVEL log_level = AMBA_LOG_INFO;
 static LOG_TARGET log_target = LOG_TARGET_STDERR;
 static char log_file[LOG_FILENAME_SIZE_MAX] = "stderr";
 
 static int print_log(const char *type, const char* func, const char *format,
-                     va_list vlist, LOG_LEVEL level)
+                     va_list vlist, AMBA_LOG_LEVEL level)
 {
 	int fd = -1;
 	char str[LOG_STRING_SIZE_MAX];
 	char fmt[LOG_FORMAT_SIZE_MAX];
-	int color = (level < LOG_INFO ? 31 : 34);
+	int color = (level < AMBA_LOG_INFO ? 31 : 34);
 	snprintf(fmt, sizeof(fmt), "\033[%dm[%s]\t%s: %s\033[39m", color, type,
 	         func, format);
 	vsnprintf(str, sizeof(str), fmt, vlist);
@@ -57,7 +75,7 @@ static int print_log(const char *type, const char* func, const char *format,
 			close(fd);
 			fd = -1;
 		}
-		if (level <= LOG_INFO)
+		if (level <= AMBA_LOG_INFO)
 			vfprintf(stderr, fmt, vlist);
 		break;
 	case LOG_TARGET_STDERR:
@@ -71,9 +89,9 @@ static int print_log(const char *type, const char* func, const char *format,
 int log_error(const char* func, const char *format, ...)
 {
 	va_list vlist;
-	if (log_level >= LOG_ERROR) {
+	if (log_level >= AMBA_LOG_ERROR) {
 	    va_start(vlist, format);
-	    print_log("ERROR", func, format, vlist, LOG_ERROR);
+	    print_log("ERROR", func, format, vlist, AMBA_LOG_ERROR);
 	    va_end(vlist);
 	}
 	return 0;
@@ -82,9 +100,9 @@ int log_error(const char* func, const char *format, ...)
 int log_info(const char* func, const char *format, ...)
 {
 	va_list vlist;
-	if (log_level >= LOG_INFO) {
+	if (log_level >= AMBA_LOG_INFO) {
 	    va_start(vlist, format);
-	    print_log("INFO", func, format, vlist, LOG_INFO);
+	    print_log("INFO", func, format, vlist, AMBA_LOG_INFO);
 	    va_end(vlist);
 	}
 	return 0;
@@ -93,9 +111,9 @@ int log_info(const char* func, const char *format, ...)
 int log_debug(const char* func, const char *format, ...)
 {
 	va_list vlist;
-	if (log_level >= LOG_DEBUG) {
+	if (log_level >= AMBA_LOG_DEBUG) {
 	    va_start(vlist, format);
-	    print_log("DEBUG", func, format, vlist, LOG_DEBUG);
+	    print_log("DEBUG", func, format, vlist, AMBA_LOG_DEBUG);
 	    va_end(vlist);
 	}
 	return 0;
@@ -104,9 +122,9 @@ int log_debug(const char* func, const char *format, ...)
 int log_trace(const char* func, const char *format, ...)
 {
 	va_list vlist;
-	if (log_level >= LOG_TRACE) {
+	if (log_level >= AMBA_LOG_TRACE) {
 	    va_start(vlist, format);
-	    print_log("TRACE", func, format, vlist, LOG_TRACE);
+	    print_log("TRACE", func, format, vlist, AMBA_LOG_TRACE);
 	    va_end(vlist);
 	}
 	return 0;
@@ -120,10 +138,10 @@ int set_log(int level, char *p_file)
 	}
 	if (level < 0) {
 	  level = 0;
-	} else if (level >= LOG_MAX) {
-	  level = LOG_MAX - 1;
+	} else if (level >= AMBA_LOG_MAX) {
+	  level = AMBA_LOG_MAX - 1;
 	}
-	log_level = (LOG_LEVEL)level;
+	log_level = (AMBA_LOG_LEVEL)level;
 	DEBUG("Set log level [%d], target [%s]\n", log_level, log_file);
 	return 0;
 }
diff --git a/make.inc b/make.inc
index 7e4d485..7cb2b21 100644
--- a/make.inc
+++ b/make.inc
@@ -1,15 +1,32 @@
 ##
-## packages/data_transfer/make.inc
+## packages/utils/make.inc
 ##
 ## History:
 ##    2012/06/01 - [Cao Rongrong] Created file
 ##
-## Copyright (C) 2012-2016, Ambarella, Inc.
+## Copyright (c) 2016 Ambarella, Inc.
 ##
-## All rights reserved. No Part of this file may be reproduced, stored
-## in a retrieval system, or transmitted, in any form, or by any means,
-## electronic, mechanical, photocopying, recording, or otherwise,
-## without the prior consent of Ambarella, Inc.
+## This file and its contents ("Software") are protected by intellectual
+## property rights including, without limitation, U.S. and/or foreign
+## copyrights. This Software is also the confidential and proprietary
+## information of Ambarella, Inc. and its licensors. You may not use, reproduce,
+## disclose, distribute, modify, or otherwise prepare derivative works of this
+## Software or any portion thereof except pursuant to a signed license agreement
+## or nondisclosure agreement with Ambarella, Inc. or its authorized affiliates.
+## In the absence of such an agreement, you agree to promptly notify and return
+## this Software to Ambarella, Inc.
+##
+## THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+## INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF NON-INFRINGEMENT,
+## MERCHANTABILITY, AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+## IN NO EVENT SHALL AMBARELLA, INC. OR ITS AFFILIATES BE LIABLE FOR ANY DIRECT,
+## INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+## (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+## LOSS OF USE, DATA, OR PROFITS; COMPUTER FAILURE OR MALFUNCTION; OR BUSINESS
+## INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+## CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+## ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+## POSSIBILITY OF SUCH DAMAGE.
 ##
 
 ifeq ($(BUILD_AMBARELLA_UTILITIES_PACKAGE), y)
diff --git a/perf.c b/perf.c
index dfc3253..38913ea 100644
--- a/perf.c
+++ b/perf.c
@@ -4,12 +4,29 @@
  * History:
  *  Jun 5, 2013 - [qianshen] created file
  *
- * Copyright (C) 2012-2016, Ambarella ShangHai Co,Ltd
+ * Copyright (c) 2016 Ambarella, Inc.
  *
- * All rights reserved. No Part of this file may be reproduced, stored
- * in a retrieval system, or transmitted, in any form, or by any means,
- * electronic, mechanical, photocopying, recording, or otherwise,
- * without the prior consent of Ambarella
+ * This file and its contents ( "Software" ) are protected by intellectual
+ * property rights including, without limitation, U.S. and/or foreign
+ * copyrights. This Software is also the confidential and proprietary
+ * information of Ambarella, Inc. and its licensors. You may not use, reproduce,
+ * disclose, distribute, modify, or otherwise prepare derivative works of this
+ * Software or any portion thereof except pursuant to a signed license agreement
+ * or nondisclosure agreement with Ambarella, Inc. or its authorized affiliates.
+ * In the absence of such an agreement, you agree to promptly notify and return
+ * this Software to Ambarella, Inc.
+ *
+ * THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF NON-INFRINGEMENT,
+ * MERCHANTABILITY, AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL AMBARELLA, INC. OR ITS AFFILIATES BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; COMPUTER FAILURE OR MALFUNCTION; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
  *
  ******************************************************************************/
 
diff --git a/utils.h b/utils.h
index a3e35d4..a99b9fa 100644
--- a/utils.h
+++ b/utils.h
@@ -4,18 +4,40 @@
  * History:
  *  Jun 5, 2013 - [qianshen] created file
  *
- * Copyright (C) 2012-2016, Ambarella ShangHai Co,Ltd
+ * Copyright (c) 2016 Ambarella, Inc.
  *
- * All rights reserved. No Part of this file may be reproduced, stored
- * in a retrieval system, or transmitted, in any form, or by any means,
- * electronic, mechanical, photocopying, recording, or otherwise,
- * without the prior consent of Ambarella
+ * This file and its contents ( "Software" ) are protected by intellectual
+ * property rights including, without limitation, U.S. and/or foreign
+ * copyrights. This Software is also the confidential and proprietary
+ * information of Ambarella, Inc. and its licensors. You may not use, reproduce,
+ * disclose, distribute, modify, or otherwise prepare derivative works of this
+ * Software or any portion thereof except pursuant to a signed license agreement
+ * or nondisclosure agreement with Ambarella, Inc. or its authorized affiliates.
+ * In the absence of such an agreement, you agree to promptly notify and return
+ * this Software to Ambarella, Inc.
+ *
+ * THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF NON-INFRINGEMENT,
+ * MERCHANTABILITY, AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL AMBARELLA, INC. OR ITS AFFILIATES BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; COMPUTER FAILURE OR MALFUNCTION; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
  *
  ******************************************************************************/
 
 #ifndef UTILS_H_
 #define UTILS_H_
 
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
 #ifndef ERROR
 #define ERROR(format, arg...)       log_error(__PRETTY_FUNCTION__, format, ##arg)
 #endif
@@ -52,12 +74,12 @@
 #endif
 
 typedef enum {
-  LOG_ERROR = 0,
-  LOG_INFO = 1,
-  LOG_DEBUG = 2,
-  LOG_TRACE = 3,
-  LOG_MAX
-} LOG_LEVEL;
+  AMBA_LOG_ERROR = 0,
+  AMBA_LOG_INFO = 1,
+  AMBA_LOG_DEBUG = 2,
+  AMBA_LOG_TRACE = 3,
+  AMBA_LOG_MAX
+} AMBA_LOG_LEVEL;
 
 int set_log(int level, char *target);
 
@@ -70,4 +92,8 @@ void start_timing(void);
 void stop_timing(void);
 void perf_report(void);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif /* UTILS_H_ */
diff --git a/version.h b/version.h
index eb0ea6d..24eb30e 100644
--- a/version.h
+++ b/version.h
@@ -1,18 +1,34 @@
 /*******************************************************************************
  * version.h
  *
- * Histroy:
+ * History:
  *  2014/07/14 2014 - [Lei Hong] created file
  *
- * Copyright (C) 2008-2014, Ambarella ShangHai Co,Ltd
+ * Copyright (c) 2016 Ambarella, Inc.
  *
- * All rights reserved. No Part of this file may be reproduced, stored
- * in a retrieval system, or transmitted, in any form, or by any means,
- * electronic, mechanical, photocopying, recording, or otherwise,
- * without the prior consent of Ambarella
+ * This file and its contents ( "Software" ) are protected by intellectual
+ * property rights including, without limitation, U.S. and/or foreign
+ * copyrights. This Software is also the confidential and proprietary
+ * information of Ambarella, Inc. and its licensors. You may not use, reproduce,
+ * disclose, distribute, modify, or otherwise prepare derivative works of this
+ * Software or any portion thereof except pursuant to a signed license agreement
+ * or nondisclosure agreement with Ambarella, Inc. or its authorized affiliates.
+ * In the absence of such an agreement, you agree to promptly notify and return
+ * this Software to Ambarella, Inc.
+ *
+ * THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF NON-INFRINGEMENT,
+ * MERCHANTABILITY, AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL AMBARELLA, INC. OR ITS AFFILIATES BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; COMPUTER FAILURE OR MALFUNCTION; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
  *
  ******************************************************************************/
-
 #ifndef AM_UTILS_VERSION_H_
 #define AM_UTILS_VERSION_H_
 
