diff --git a/AmbaConfig b/AmbaConfig
index 41d7bd9..2231ce0 100644
--- a/AmbaConfig
+++ b/AmbaConfig
@@ -4,12 +4,29 @@
 ## History:
 ##    2009/07/08 - [Anthony Ginger]
 ##
-## Copyright (C) 2004-2008, Ambarella, Inc.
+## Copyright (c) 2016 Ambarella, Inc.
 ##
-## All rights reserved. No Part of this file may be reproduced, stored
-## in a retrieval system, or transmitted, in any form, or by any means,
-## electronic, mechanical, photocopying, recording, or otherwise,
-## without the prior consent of Ambarella, Inc.
+## This file and its contents ("Software") are protected by intellectual
+## property rights including, without limitation, U.S. and/or foreign
+## copyrights. This Software is also the confidential and proprietary
+## information of Ambarella, Inc. and its licensors. You may not use, reproduce,
+## disclose, distribute, modify, or otherwise prepare derivative works of this
+## Software or any portion thereof except pursuant to a signed license agreement
+## or nondisclosure agreement with Ambarella, Inc. or its authorized affiliates.
+## In the absence of such an agreement, you agree to promptly notify and return
+## this Software to Ambarella, Inc.
+##
+## THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+## INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF NON-INFRINGEMENT,
+## MERCHANTABILITY, AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+## IN NO EVENT SHALL AMBARELLA, INC. OR ITS AFFILIATES BE LIABLE FOR ANY DIRECT,
+## INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+## (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+## LOSS OF USE, DATA, OR PROFITS; COMPUTER FAILURE OR MALFUNCTION; OR BUSINESS
+## INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+## CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+## ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+## POSSIBILITY OF SUCH DAMAGE.
 ##
 
 config BUILD_AMBARELLA_TEXTINSERT_PACKAGE
diff --git a/make.inc b/make.inc
index c597570..e782375 100644
--- a/make.inc
+++ b/make.inc
@@ -4,12 +4,29 @@
 ## History:
 ##    2012/06/01 - [Cao Rongrong] Created file
 ##
-## Copyright (C) 2012-2016, Ambarella, Inc.
+## Copyright (c) 2016 Ambarella, Inc.
 ##
-## All rights reserved. No Part of this file may be reproduced, stored
-## in a retrieval system, or transmitted, in any form, or by any means,
-## electronic, mechanical, photocopying, recording, or otherwise,
-## without the prior consent of Ambarella, Inc.
+## This file and its contents ("Software") are protected by intellectual
+## property rights including, without limitation, U.S. and/or foreign
+## copyrights. This Software is also the confidential and proprietary
+## information of Ambarella, Inc. and its licensors. You may not use, reproduce,
+## disclose, distribute, modify, or otherwise prepare derivative works of this
+## Software or any portion thereof except pursuant to a signed license agreement
+## or nondisclosure agreement with Ambarella, Inc. or its authorized affiliates.
+## In the absence of such an agreement, you agree to promptly notify and return
+## this Software to Ambarella, Inc.
+##
+## THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+## INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF NON-INFRINGEMENT,
+## MERCHANTABILITY, AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+## IN NO EVENT SHALL AMBARELLA, INC. OR ITS AFFILIATES BE LIABLE FOR ANY DIRECT,
+## INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+## (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+## LOSS OF USE, DATA, OR PROFITS; COMPUTER FAILURE OR MALFUNCTION; OR BUSINESS
+## INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+## CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+## ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+## POSSIBILITY OF SUCH DAMAGE.
 ##
 
 ifeq ($(BUILD_AMBARELLA_TEXTINSERT_PACKAGE), y)
diff --git a/text_insert.c b/text_insert.c
index ad9a12d..2cbc5d3 100644
--- a/text_insert.c
+++ b/text_insert.c
@@ -1,16 +1,34 @@
-/*
+/*******************************************************************************
  * text_insert.c
  *
  * History:
  *  2011/02/25 - [Qian Shen] created file
- * Copyrigt (C) 2007-2011, Ambarella, Inc
  *
- *  All rights reserved. No Part of this file may be reproduced, stored in a
- *  retrieval system, or transmitterd, in any form, or any means,
- *  electronic, mechanical, photocopying, recording, or otherwise, without the
- *  prior consent of Ambarella, Inc.
+ * Copyright (c) 2016 Ambarella, Inc.
  *
- */
+ * This file and its contents ( "Software" ) are protected by intellectual
+ * property rights including, without limitation, U.S. and/or foreign
+ * copyrights. This Software is also the confidential and proprietary
+ * information of Ambarella, Inc. and its licensors. You may not use, reproduce,
+ * disclose, distribute, modify, or otherwise prepare derivative works of this
+ * Software or any portion thereof except pursuant to a signed license agreement
+ * or nondisclosure agreement with Ambarella, Inc. or its authorized affiliates.
+ * In the absence of such an agreement, you agree to promptly notify and return
+ * this Software to Ambarella, Inc.
+ *
+ * THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF NON-INFRINGEMENT,
+ * MERCHANTABILITY, AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL AMBARELLA, INC. OR ITS AFFILIATES BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; COMPUTER FAILURE OR MALFUNCTION; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************/
 
 #include <stdio.h>
 #include <unistd.h>
@@ -633,6 +651,7 @@ int text2bitmap_convert_character(u32 ui_char_code,
 	int ft_error;
 	int width, height, left, top, advance_x,advance_y, bmp_pitch = 0;
 	u8 *mapped_pixel = NULL;
+	int rtn_val = 0;
 	FT_GlyphSlot slot = face->glyph;
 
 	if (buf_height < cfg_font.size) {
@@ -690,6 +709,7 @@ int text2bitmap_convert_character(u32 ui_char_code,
 	if (height > buf_height) {
 		printf("Error: converted bitmap height [%d] exceeds the buffer height [%d]\n",
 			height, buf_height);
+		rtn_val = -1;
 		goto ERROR_QUIT;
 	}
 
@@ -699,13 +719,12 @@ int text2bitmap_convert_character(u32 ui_char_code,
 	if ((bmp_pitch > 0) && (bmp_addr != NULL)) {
 		int row = 0, col = 0, dst_index = 0, src_index = 0;
 		u8 * ptr = NULL;
-		u32 fond_gap = cfg_font.size - top;
-		u32 pop_adjust = cfg_font.size - height;
+		u32 font_gap = cfg_font.size - top;
 	//	printf("\n\n");
-		if (((fond_gap + height) > buf_height) && (fond_gap > pop_adjust))
-			fond_gap = buf_height - height;
+		if ((font_gap + height) > buf_height)
+			font_gap = buf_height - height;
 		for (row = 0; row < height; row++) {
-			dst_index = buf_pitch * (row + fond_gap) + offset_x;
+			dst_index = buf_pitch * (row + font_gap) + offset_x;
 			src_index = row * bmp_pitch;
 			for (col = 0; col < width; col++, dst_index++, src_index++) {
 				if (dst_index < 0)
@@ -733,7 +752,7 @@ ERROR_QUIT:
 		free(mapped_pixel);
 		mapped_pixel = NULL;
 	}
-	return 0;
+	return rtn_val;
 }
 
 /*
@@ -741,14 +760,12 @@ ERROR_QUIT:
  */
 int text2bitmap_lib_deinit()
 {
-	if (flag_init == 0) {
-		printf("text2bitmap library hasn't been initilized.");
-		return 0;
+	if (flag_init == 1) {
+		FT_Done_Face(face);
+		FT_Done_FreeType(library);
+		flag_init = 0;
+		flag_face = 0;
 	}
-	FT_Done_Face(face);
-	FT_Done_FreeType(library);
-	flag_init = 0;
-	flag_face = 0;
 	return 0;
 }
 
diff --git a/text_insert.h b/text_insert.h
index db6e654..a3b7ca3 100644
--- a/text_insert.h
+++ b/text_insert.h
@@ -1,16 +1,34 @@
-/*
+/*******************************************************************************
  * text_insert.h
  *
  * History:
  *  2011/02/25 - [Qian Shen] created file
- * Copyrigt (C) 2007-2011, Ambarella, Inc
  *
- *  All rights reserved. No Part of this file may be reproduced, stored in a
- *  retrieval system, or transmitterd, in any form, or any means,
- *  electronic, mechanical, photocopying, recording, or otherwise, without the
- *  prior consent of Ambarella, Inc.
+ * Copyright (c) 2016 Ambarella, Inc.
  *
- */
+ * This file and its contents ( "Software" ) are protected by intellectual
+ * property rights including, without limitation, U.S. and/or foreign
+ * copyrights. This Software is also the confidential and proprietary
+ * information of Ambarella, Inc. and its licensors. You may not use, reproduce,
+ * disclose, distribute, modify, or otherwise prepare derivative works of this
+ * Software or any portion thereof except pursuant to a signed license agreement
+ * or nondisclosure agreement with Ambarella, Inc. or its authorized affiliates.
+ * In the absence of such an agreement, you agree to promptly notify and return
+ * this Software to Ambarella, Inc.
+ *
+ * THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF NON-INFRINGEMENT,
+ * MERCHANTABILITY, AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL AMBARELLA, INC. OR ITS AFFILIATES BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; COMPUTER FAILURE OR MALFUNCTION; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************/
 
 #ifndef _TEXT2BITMAP_LIB_H
 #define _TEXT2BITMAP_LIB_H
